apps/authentication/models.py

```diff
@@ -67,6 +80,36 @@ def validate_category(self, key, value):
         if value not in allowed:
             raise ValueError(f"Invalid user category: {value} -- {allowed}")
         return value
+    
+
+class UserGroups(db.Model):
+    __tablename__ = 'user_groups'
+    user_id = db.Column(db.Integer, db.ForeignKey("users.id"), primary_key=True)
+    group_id = db.Column(db.Integer, db.ForeignKey("groups.id"), primary_key=True)
+    user = db.relationship("Users", back_populates="user_groups", foreign_keys=[user_id])
+    group = db.relationship("Groups", back_populates="user_groups")
+
+
+    owner_id = db.Column(db.Integer, db.ForeignKey("users.id"), nullable=True)
+    assistant_id = db.Column(db.Integer, db.ForeignKey("users.id"), nullable=True)
+    member_id = db.Column(db.Integer, db.ForeignKey("users.id"), nullable=True)
+
+
+    owner = db.relationship(
+        "Users",
+        foreign_keys=[owner_id],
+        remote_side="Users.id"
+    )
+    assistant = db.relationship(
+        "Users",
+        foreign_keys=[assistant_id],
+        remote_side="Users.id"
+    )
+    member = db.relationship(
+        "Users",
+        foreign_keys=[member_id],
+        remote_side="Users.id"
+    )
```

Comment:
Can you please double check here if this is the best approach to keep track of the "type" of membership an user can have?

The way it was designed above, the table will ended up have at least two empty columns for each member, and the user_id will be duplicated with either owner_id, assistant_id or member_id.

The way I picture this would be something like: an enum constant that specifies the "type of membership" (let's say: 1 - member; 2 - assistant; 3 - owner), and then this type of membership becomes another field that will compose the primary keys on the table.